port/unix/omrsock.c

```diff
@@ -31,10 +31,143 @@
 
 #include "omrcfg.h"
 #if defined(OMR_PORT_SOCKET_SUPPORT)
+#include "omrsock.h"
 #include "omrport.h"
 #include "omrporterror.h"
 #include "omrsockptb.h"
 
+/* Internal: OMRSOCK user interface constants TO OS dependent constants mapping. */
+
+/**
+ * @internal Map OMRSOCK API user interface address family names to the OS address 
+ * family, which may be defined differently depending on operating system. 
+ *
+ * @param OMRAddrFamily The OMR address family to be converted.
+ *
+ * @return OS address family, or OS_SOCK_AF_UNSPEC if none exists.
+ */
+static int32_t
+map_addr_family_OMR_to_OS(int32_t OMRAddrFamily)
+{
+	switch(OMRAddrFamily)
+	{
+		case OMRSOCK_AF_INET:
+			return OS_SOCK_AF_INET;
+		case OMRSOCK_AF_INET6:
+			return OS_SOCK_AF_INET6;
+	}
+	return OS_SOCK_AF_UNSPEC;
+}
+
+/**
+ * @internal Map OMRSOCK API user interface socket type to the OS socket type, 
+ * which may be defined differently depending on operating system. 
+ *
+ * @param OMRSockType The OMR socket type to be converted.
+ *
+ * @return OS socket type on success, or OS_SOCK_ANY if none exists.
+ */
+static int32_t
+map_sockettype_OMR_to_OS(int32_t OMRSockType)
+{
+	switch(OMRSockType)
+	{
+		case OMRSOCK_STREAM:
+			return OS_SOCK_STREAM;
+		case OMRSOCK_DGRAM:
+			return OS_SOCK_DGRAM;
+	}
+	return OS_SOCK_ANY;
+}
+
+/**
+ * @internal Map OMRSOCK API user interface protocol to the OS protocols, 
+ * which may be defined differently depending on operating system. 
+ *
+ * @param OMRProtocol The OMR protocol to be converted.
+ *
+ * @return OS protocol on success, or OS_SOCK_IPPROTO_DEFAULT 
+ * if none exists.
+ */
+static int32_t
+map_protocol_OMR_to_OS(int32_t OMRProtocol)
```

Comment:
1. `OMRProtocol` is named like a type, it should be written `omrProtocol`.
2. I don't like the mixed case in the function name, I would rather it be `map_protocol_omr_to_os`, or even change the name to something simpler, `get_os_protocol`.

Reply:
These comments apply to all the mapper functions, not just this one.

Reply:
Sounds good! I think changing it to get_os_protocol and such sounds good, and it is more simplistic and easier to call the function too.

Reply:
I also changed the argument name.